fetch_submissions.py

```diff
@@ -40,19 +22,38 @@ def progress(count, total, status=''):
 
 
 def get_file_extension(file_type):
-  if file_type in FILE_EXTENSIONS:
-    return FILE_EXTENSIONS[file_type]
-  return 'txt'
-
-
-def get_comment_symbol(file_type):
-  if file_type in ['sql']:
-    return '--'
-  if file_type in ['python3', 'python', 'php', 'ruby', 'bash']:
-    return '#'
-  if file_type in ['csharp', 'rust', 'kotlin', 'scala', 'golang', 'swift',
-                   'javascript', 'c', 'java', 'cpp']:
-    return '//'
+  try:
+    return {
+      'mysql': 'sql',
+      'bash': 'sh',
+      'python3': 'py',
+      'php': 'php',
+      'rust': 'rs',
+      'kotlin': 'kt',
+      'scala': 'scala',
+      'golang': 'go',
+      'swift': 'swift',
+      'ruby': 'rb',
+      'javascript': 'py',
+      'csharp': 'cs',
+      'c': 'c',
+      'python': 'py',
+      'java': 'java',
+      'cpp': 'cpp'
+    }[file_type]
+  except KeyError:
+    return 'txt'
```

Comment:
Since the script has mappings for all 16 languages supported on LeetCode to their file extensions, the case where the submission language returned cannot be found is an exceptional case. Hence I've added an `try-except` clause to handle the conversion from file type to a file extension.

Reply:
I'm fine with the `try-except` but why do you want to move the dictionary down into the function? It would just construct the dictionary whenever the function is called unnecessarily. Also, an alternative is using `defaultdict`.

Reply:
Hmm it shouldn't be that expensive an operation, I wanted to mainly remove a global dependency. I think using a `defaultdict` is a good option.

Reply:
Why do you want to remove it?

Reply:
The dictionary is used only within that function so I don't think it needs to be declared globally.

Reply:
Anyways I had a look at the `defaultdict` object in the `collections` module. I don't think that suits this particular use case. From what I understood, it can be used to set default values for the keys in the dictionary. I think you meant to suggest something like `LazyDict` (https://pypi.org/project/lazydict/) as an alternative so that the mapping is not reconstructed everytime the function is called. I think it's an overkill for this situation because the dictionary contains only 16 key value pairs so the time spent in construction is negligible as compared to say the download of the solution files.

Reply:
If you move the dictionary into the function, it is recreated at each function call. And this is a rather frequently called function. It makes sense to have it outside so that the work of creating the dictionary is not replicated so many times.

A `defaultdict` will save you the logic of `try-except`. Here's an example. Say we only support python, Java and C++:

```python
file_extensions = defaultdict(lambda: 'txt', {
    'python': 'py',	
    'java': 'java',	
    'cpp': 'cpp'
})

file_extensions['python'] # returns 'py'
file_extensions['java']   # returns 'java'
file_extensions['go']     # returns 'txt'
```

Reply:
So you still want the `defaultdict` object to be created globally right? I mean otherwise we'll still be recreating the `defaultdict` object with every function call, same as the dictionary.

Reply:
Yup, that's right!